web/src/beta/features/Editor/Story/PagesPanel/PageItem.tsx

```diff
@@ -1,141 +1,133 @@
+  const [localTitle, setLocalTitle] = useState(title);
 
-  const handleTitleSubmit = useCallback(
-    (newTitle: string) => {
-      setIsEditing(false);
-      setNewValue(newTitle);
-      if (newValue?.trim() !== "") {
-        const schemaGroupId = storyPage.property.items?.[1]?.schemaGroup;
-        onPropertyUpdate?.(propertyId, schemaGroupId, "title", undefined, "string", newTitle);
-      }
-    },
-    [newValue, onPropertyUpdate, propertyId, storyPage.property.items],
-  );
+  const handleStoryPageItemClick = useCallback(() => {
+    if (storyPage.id === selectedStoryPage?.id) return;
+    handleStoryPageSelect(storyPage.id);
+    setEditingPageNameId("");
+  }, [storyPage.id, selectedStoryPage?.id, handleStoryPageSelect]);
+
+  const handleTitleUpdate = useCallback(() => {
+    setEditingPageNameId("");
+    if (!localTitle || localTitle === storyPage.title) return;
+    const schemaGroupId = storyPage.property.items?.[1]?.schemaGroup;
+    handlePropertyValueUpdate?.(
+      storyPage.property.id,
+      schemaGroupId,
+      "title",
+      undefined,
+      "string",
+      localTitle,
+    );
+  }, [
+    handlePropertyValueUpdate,
+    localTitle,
+    storyPage.property.id,
+    storyPage.property.items,
+    storyPage.title,
+  ]);
 
-  const handleEditExit = useCallback(
-    (e?: React.KeyboardEvent<HTMLInputElement>) => {
-      if (!!newValue && title !== newValue && e?.key !== "Escape") {
-        handleTitleSubmit(newValue);
-      } else {
-        setNewValue(title);
-      }
-      setIsEditing(false);
-    },
-    [title, newValue, handleTitleSubmit],
+  const optionsMenu = useMemo(
+    () => [
+      {
+        id: "rename",
+        title: "Rename",
+        icon: "pencilSimple" as const,
+        onClick: () => setEditingPageNameId(storyPage.id),
+      },
+      {
+        id: "delete",
+        title: "Delete",
+        icon: "trash" as const,
+        onClick: () => handleStoryPageDelete(storyPage.id),
+      },
+    ],
+    [storyPage.id, handleStoryPageDelete],
   );
 
   return (
-    <ListItem
-      border
-      actionPlacement="bottom-start"
-      isSelected={isSelected}
-      isOpenAction={isOpenAction}
-      onItemClick={handleSingleClick}
-      onActionClick={onActionClick}
-      onOpenChange={onOpenChange}
-      actionContent={
-        <PopoverMenuContent
-          size="sm"
-          items={[
-            // {
-            //   icon: "copy",
-            //   name: "Duplicate",
-            //   onClick: () => {
-            //     setOpenedPageId(undefined);
-            //     onPageDuplicate(storyPage.id);
-            //   },
-            // },
-            {
-              icon: "trash",
-              name: "Delete",
-              onClick: () => {
-                setOpenedPageId?.(undefined);
-                onPageDelete?.();
-              },
-            },
-          ]}
+    <Wrapper>
+      <Typography size="body">{pageCount}.</Typography>
```

Comment:
Please doube confirm with Red about the design, this Typography could have different width when it goes to two numbers (or more), and making the entry item not aligned vertically. Maybe better have a min width?
![image](https://github.com/reearth/reearth-visualizer/assets/21994748/489e5920-816d-49ba-9c26-e697383ec10d)